src/SFA.DAS.LevyTransferMatching.Web/Orchestrators/OpportunitiesOrchestrator.cs

```diff
@@ -1,18 +1,113 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
 using System.Linq;
 using System.Threading.Tasks;
+using SFA.DAS.LevyTransferMatching.Infrastructure.Services.DateTimeService;
 using SFA.DAS.LevyTransferMatching.Infrastructure.Services.OpportunitiesService;
+using SFA.DAS.LevyTransferMatching.Infrastructure.Services.TagService;
+using SFA.DAS.LevyTransferMatching.Web.Extensions;
 using SFA.DAS.LevyTransferMatching.Web.Models.Opportunities;
 
 namespace SFA.DAS.LevyTransferMatching.Web.Orchestrators
 {
     public class OpportunitiesOrchestrator : IOpportunitiesOrchestrator
     {
+        private const string All = "All";
+
+        private readonly IDateTimeService _dateTimeService;
         private readonly IOpportunitiesService _opportunitiesService;
+        private readonly ITagService _tagService;
 
-        public OpportunitiesOrchestrator(IOpportunitiesService opportunitiesService)
+        public OpportunitiesOrchestrator(IDateTimeService dateTimeService, IOpportunitiesService opportunitiesService, ITagService tagService)
         {
+            _dateTimeService = dateTimeService;
             _opportunitiesService = opportunitiesService;
+            _tagService = tagService;
+        }
+
+        public async Task<DetailViewModel> GetDetailViewModel(string encodedId)
+        {
+            var opportunityDto = await _opportunitiesService.GetOpportunity(encodedId);
+
+            // If null, an opportunity couldn't be found for the encodedId.
+            if (opportunityDto == null)
+            {
+                return null;
+            }
+
+            // Pull back the tags, and use the descriptions to build the lists.
+            var sectorTags = await _tagService.GetSectors();
```

Comment:
All this feels like something we could do elsewhere and make it re-usable

Reply:
I thought that and toyed around with the idea of making it common. Decided against it given Levels are treated differently to Sectors and JobRoles. I was thinking that, if we come across the need to re-use this functionality, we could always do it then (and the same for the special case of Levels too).

Your thoughts?

Reply:
Putting Levels aside for a moment, which is awkward and may end up being a bit inelegant, there's still a lot of code here for Sectors and JobRoles, and surely that could be tucked away elsewhere.

There's no _if_ about the need to re-use - all of the next set of stories (creating a pledge application) will need to show this exact same thing.

I think we have time to rework this, maybe even into a partial, so that it's easy to reuse.

Reply:
If this is going to be needed, I guess there's no harm in refactoring early. I'll get to work on this today.

Reply:
I've created a new partial, `_OpportunitySummary.cshtml`, along with it's own view model, and seperated out the logic to populate this view model in the `OpportunitiesOrchestrator`, so that this can be reused by other views.